articles/openshift/howto-deploy-java-eap-app.md

```diff
@@ -0,0 +1,384 @@
+    ```
+
+1. Monitor the status of the pods and wait until the database server is running:
+
+    ```bash
+    todo-list (bootable-jar-openshift *) $ oc get pods -w
+    NAME                   READY   STATUS      RESTARTS   AGE
+    mssqlserver-1-deploy   0/1     Completed   0          34s
+    mssqlserver-1-gw7qw    1/1     Running     0          31s
+    ```
+
+1. Connect to the database pod and create the database `todos_db`:
+
+    ```bash
+    todo-list (bootable-jar-openshift) $ oc rsh mssqlserver-1-gw7qw
+    sh-4.4$ /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P 'Passw0rd!'
+    1> CREATE DATABASE todos_db
+    2> GO
+    1> exit
+    sh-4.4$ exit
+    exit
+    ```
+
+### Deploy the application to OpenShift
+
+Now that we have the database server ready, we can deploy the demo application via JBoss EAP Helm Charts. The Helm Chart application configuration file is available at `todo-list/deployment/application/todo-list-helm-chart.yaml`. You could deploy this file via command line, however, to do so you would need to have Helm Charts installed on your local machine. Instead of using the command line, the next steps explain how you can deploy this Helm Chart by using the OpenShift web console.
+
+Before deploying the application, let's creating the expected secret that will hold specific application configuration. The Helm Chart will get the database user, password and name from the `mssqlserver-secret`secret object created before, and the driver version, the datasource JNDI name and the cluster password from the following secret:
+
+1. Execute the following to create the application config map:
+
+    ```bash
+    todo-list (bootable-jar-openshift) $ oc create secret generic todo-list-secret \
+    --from-literal app-driver-version=7.4.1.jre11 \
+    --from-literal app-ds-jndi=java:/comp/env/jdbc/mssqlds \
+    --from-literal app-cluster-password=mut2UTG6gDwNDcVW
+    ```
+
+    > [!NOTE]
+    > You decide which cluster password you want to use, the pods that want to join to your cluster need such password, that prevents your JBoss EAP cluster is not messed up with any pod that are not under your control and ask to join.
+
+    > [!NOTE]
+    > You could have noticed from the above config map that we are not supplying the database Hostname and Port. That's not necessary. If you take a closer look at the application Helm Chart file, you will see that the database Hostname and Port are passed by using the following notations \$(MSSQLSERVER_SERVICE_HOST) and \$(MSSQLSERVER_SERVICE_PORT). This is a standard OpenShift notation that will make the application variables (MSSQLSERVER_HOST, MSSQLSERVER_PORT) get assigned to the values of pod environment variables (MSSQLSERVER_SERVICE_HOST, MSSQLSERVER_SERVICE_PORT) available at runtime, which are passed by OpenShift when the pod is launched. These variables are available to any pod because we have created a Service to expose the SQL server in the previous steps.
```

Comment:
s/could have noticed/may have noticed/g

s/notation that will make/notation that will ensure/g

s/values of pod/values of the pod/g

s/available/that are available/g

s/, which are passed/. Pod environment variable are passed/g